src/hotspot/share/gc/g1/g1CollectionSetCandidates.cpp

```diff
@@ -35,6 +35,27 @@ void G1CollectionSetCandidates::remove(uint num_regions) {
   }
 }
 
+void G1CollectionSetCandidates::prune(uint keep_min_regions, size_t prune_total_bytes) {
+  uint regions_left = _num_regions;
+  size_t pruned_bytes = 0;
+
+  while (regions_left > keep_min_regions) {
+    uint cur_idx = regions_left - 1;
+    // Do not prune more than prune_total_bytes.
+    if ((at(cur_idx)->reclaimable_bytes() + pruned_bytes) <= prune_total_bytes) {
```

Comment:
This check should be the other way around, right?
```suggestion
    if ((at(cur_idx)->reclaimable_bytes() + pruned_bytes) > prune_total_bytes) {
```